### ISSUE TITLE
User is maybe able to monetize localhost

### ISSUE THREAD

**Issue by kuopassa** [moneystreamdev/moneystream-project#3]
The MoneyStream icon turns green (with Chrome) when I'm on localhost developing a site, such as `http://localhost/bsv.tips/`. Maybe allowing users to monetize localhost is an issue if someone wants to game the system somehow.

**Comment by dfoderick** [comment]
I think localhost content should be monetized, if only for local testing. Let me know if you have a specific reason why it should be turned off.

Are you thinking about a case where someone downloads a web site then removes monetization or they change the paymail? This is always possible for any content. Someone could "bootleg" a monetized movie using screen recording and then post the bootleg video on bittorrent. The main point of monetization is to make it easy for people to pay and most people will pay.

I think there might be some times when monetization needs to be controlled. I will leave this issue open so you can think of some reasons to control or prevent the monetization then add it here.

So far, most sites have just been adding the meta tag. This works, but users can easily see that they can still get the content even if they stop paying. Monetized web sites need to get more sophisticated by using `monetizationprogress` and `monetizationstop` events to control showing and hiding premium content. I need to provide better examples how to use the monetization events.

Bottom line, yes, there needs to be control so users cannot game the system. Most of those techniques already exist. I need to start pointing that out to people.

**Comment by kuopassa** [comment]
Thanks for the comprehensive answer. :-) I hadn't thought it through, but was thinking about if someone creates **something** that can automate the monetization without actually viewing/reading/listening the content. So for example if at some point in time there would appear the equivalent of a "click farm", then blocking that would be desirable. To block something like that is probs easiest with domain name or HTTP refe(r)rer.

**Comment by dfoderick** [comment]
Yes, for the case when user gets paid for watching ads. I had to add some code to my "video faucet" onboarding tool because someone was abusing it. I had to check ip address and only allow max 3 payouts to an ip address with a 3 hour interval between payouts. You can imagine other techniques to make sure monetization does not get abused. And just don't keep too much money in one wallet.
